app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt

```diff
@@ -307,7 +307,13 @@ class HomeActivityTest {
       profileId = profileId1,
       timestampOlderThanOneWeek = false
     )
+    val loginFirstTimeData = profileTestHelper.logIntoUser()
+    val loginSecondTimeData = profileTestHelper.logIntoUser()
     launch<HomeActivity>(createHomeActivityIntent(internalProfileId1)).use {
+      it.onActivity {
+        profileTestHelper.waitForOperationToComplete(loginFirstTimeData)
+        profileTestHelper.waitForOperationToComplete(loginSecondTimeData)
+      }
```

Comment:
I don't think you need this or the onActivity below--ProfileTestHelper should ensure that the login is completed by the time it returns.

Reply:
Has this been resolved?

Reply:
Yes, according to what we discussed in the meeting, this was made a separate function and dataMonitorFactory was used instead of the legacy code.